src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java

```diff
@@ -121,38 +133,51 @@ public PackageObjectFactory(String packageName, ClassLoader moduleClassLoader) {
      */
     @Override
     public Object createModule(String name) throws CheckstyleException {
-        Object instance = createObjectFromMap(name);
+        Object instance = null;
+        // if the name is a simple class name, try to find it in maps at first
+        if (!name.contains(".")) {
+            instance = createObjectFromMap(name, NAME_TO_FULL_MODULE_NAME);
+            if (instance == null) {
+                if (thirdPartyNameToFullModuleName == null) {
+                    thirdPartyNameToFullModuleName =
+                            generateThirdPartyNameToFullModuleName(moduleClassLoader);
```

Comment:
@rnveach , I not following you, please explain what you want to share here  with user ?